planner/cascades/transformation_rules.go

```diff
@@ -592,10 +646,17 @@ func (r *PushSelDownJoin) OnTransform(old *memo.ExprIter) (newExprs []*memo.Grou
 		join.RightJoinKeys = append(join.RightJoinKeys, eqCond.GetArgs()[1].(*expression.Column))
 	}
 	// TODO: Update EqualConditions like what we have done in the method join.updateEQCond() before.
-	leftGroup = buildChildSelectionGroup(sel, leftCond, joinExpr.Children[0])
-	rightGroup = buildChildSelectionGroup(sel, rightCond, joinExpr.Children[1])
+	leftGroup = buildChildSelectionGroup(sel, leftCond, leftGroup)
+	rightGroup = buildChildSelectionGroup(sel, rightCond, rightGroup)
 	newJoinExpr := memo.NewGroupExpr(join)
 	newJoinExpr.SetChildren(leftGroup, rightGroup)
+	if len(remainCond) > 0 {
+		sel.Conditions = remainCond
```

Comment:
This is a inplace change?

Reply:
Yes.

Reply:
We need a sel to represent the remained filters.
The old sel should keep unchanged.

Reply:
What’s the matter if we change the old selection to a new one?

Reply:
Suppose that the `sel` of `sel->join` is the child of `union` and the `sel` is pushed down to `union`. 
Since we don't need to change anything when we push the sel down to union. So though the `sel` is pushed down to every child of union. We don't make a copy of it since nothing is changed. 
Then you change it here. It would affect the `sel`s of `union`'s other children.

Reply:
In this case. It makes sense to copy selection when push down union, in my opinion. But for safety, I will generate a new selection here.

Reply:
It will be more strange that: nothing is changed but you copy it. things are changed but you don't copy it.